File: mycollab-web/src/main/scala/com/mycollab/module/project/event/TicketEvent.scala
package com.mycollab.module.project.event

import com.mycollab.events.ApplicationEvent
import com.mycollab.module.project.domain.criteria.ProjectTicketSearchCriteria

import scala.beans.BeanProperty

/**
  * @author MyCollab Ltd
  * @since 5.4.3
  */
object TicketEvent {
  
  class GotoDashboard(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
  
  class NewTicketAdded(source: AnyRef, @BeanProperty val typeVal: String, @BeanProperty val typeIdVal: Integer) extends
    ApplicationEvent(source, null) {}
  
  class SearchRequest(source: AnyRef, data: ProjectTicketSearchCriteria) extends ApplicationEvent(source, data) {}
  
  class HasTicketPropertyChanged(source: AnyRef, data: String) extends ApplicationEvent(source, data) {}
  
}

File: mycollab-web/src/main/java/com/mycollab/module/project/event/TicketEvent.kt
package com.mycollab.module.project.event

import com.mycollab.vaadin.event.ApplicationEvent
import com.mycollab.module.project.domain.criteria.ProjectTicketSearchCriteria

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object TicketEvent {
    class GotoDashboard(source: Any, val data: Any?) : ApplicationEvent(source)

    class NewTicketAdded(source: Any, val typeVal: String, val typeIdVal: Int) : ApplicationEvent(source)

    class SearchRequest(source: Any, data: ProjectTicketSearchCriteria) : ApplicationEvent(source)

    class HasTicketPropertyChanged(source: Any, data: String) : ApplicationEvent(source)
}